controller.py

```diff
@@ -0,0 +1,103 @@
+import typing
+
+import serial
+import enum
+import logging
+
+import util
+from state import State, Error
+
+
+LOGGER = logging.getLogger('controller')
+
+class StateKeys:
+    CURR_X = 'curr_x'
+    TRGT_X = 'trgt_x'
+    CURR_V = 'curr_v'
+    TRGT_V = 'trgt_v'
+    CURR_A = 'curr_a'
+    TRGT_A = 'trgt_a'
+    POLE_ANG = 'pole_ang'
+    POLE_VEL = 'pole_vel'
+    TIMESTAMP = 'timestamp'
+    ERRCODE = 'errcode'
+
+
+class ActionType(enum.Enum):
+    PositionControl = 1
+    VelocityControl = 2
+    AccelerationControl = 3
+
+
+class Action:
+    def __init__(self, action_type: ActionType, value: float) -> None:
+        self.type = action_type
+        self.value = value
+
+    def __str__(self) -> str:
+        action_str = {
```

Comment:
Create a hash map at each function call is bad idea.

Reply:
fixed hash map

About more specific name:
Currently we have and use only one string representation of action, which is the interface representation, i.e. 'TRGT_X=0.5'. Adding more specific function name here does not clarify anything as there is nothing to clarify about action string representation, but it adds unnecessary clutter in the code -- calling another method instead of doing nothing. Therefore I don't think that any change here is needed, but maybe it will be some day (when we will have more than one textual representation of action)